README.md

```diff
@@ -0,0 +1,713 @@
+/* data =>
+{
+    "de": {
+        "metadata": {...},
+        "data: {
+            "ü": {...}
+        }
+    },
+    "es": {
+        "metadata": {...},
+        "data": {
+            "ú": {...},
+            "ü": {...}
+        }
+    }
+}
+*/
+```
+
+If no matching bases are found in the database, an object with an error "message" is returned.
+
+```js
+var diacritics = require("diacritics-transliterator");
+var test = diacritics.getBase(["&"]);
+// test => { "message": "No matching bases found" }
+```
+
+### diacritics.getDecompose(array)
+
+Returns the decompose data for the matching decompose values within the array.
+
+The `array` parameter contains either an `array` of diacritic decompose string, or a `string` containing a single decompose string (internally converted into an array), of characters to process.
+
+The returned value will be an object containing all languages that match each diacritic decompose value, but the language data will only contain the matching diacritic.
+
+```js
+var diacritics = require("diacritics-transliterator");
+var data = diacritics.getDecompose(["ss"]);
+/* data =>
+{
+    "de": {
+        "metadata": {...},
+        "data: {
+            "ß": {...}
+        }
+    }
+}
+*/
+```
+
+If no matching decomposes are found in the database, an object with an error "message" is returned.
+
+```js
+var diacritics = require("diacritics-transliterator");
+var test = diacritics.getDecompose(["&", "test"]);
+// test => { "message": "No matching decomposes found" }
+```
+
+## Transliteration functions
+
+### diacritics.transliterate(string [, type][, variant])
+
+Replaces diacritics within the string with either the base or decomposed value.
+
+- The `string` parameter (type: `string`) will replace any diacritic characters with base or decomposed values. Any non-diacritics characters will not be modified.
+- The `type` parameter (type: `string`) is set to either "base" or "decompose". Defaults to "base". If a "base" value is not found, a "decompose" value will be used instead; and vice-versa.
+- The `variant` optional parameter (type: `string`) is set using the language variant. If no variant is provided, the first matching language for the diacritic will be used.
+
+The returned value will be a string with all matching diacritics replaced with the set "base" or "decompose" value.
+
+**Note** that if a `variant` parameter is set and the string contains a diacritic not found in that language variant, it will not be processed!
+
+```js
+var transliterate = require("diacritics-transliterator").transliterate;
+var german = transliterate("¿abcñ-ß123?", "decompose" "de"); // "¿abcñ-ss123?"
+var spanish = transliterate("¿abcñ-ß123?", "base", "es"); // "?abcn-ß123?"
+// The base of ß is still ß
+var generic = transliterate("¿abcñ-ß123?"); // "?abcn-ß123?"
+// If no diacritics match the set language variant, the original string is returned
+var unchanged = transliterate("¿abcñ-ß123?", "base", "test"); // "¿abcñ-ß123?"
```

Comment:
General question: How are exceptions handled? I mean "test" is invalid, you could also expect an error. The same applies to an invalid value instead of "base" or "decompose".